OSDOCS#6020: cert-manager: Customizing API fields (#59726)
openshift/openshift-docs: modules/cert-manager-override-environment-variables.adoc

// Module included in the following assemblies:
//
// * security/cert_manager_operator/cert-manager-customizing-api-fields.adoc

:_content-type: PROCEDURE
[id="cert-manager-override-environment-variables_{context}"]
= Customizing cert-manager by overriding environment variables from the cert-manager Operator API

You can override the default environment variables for the {cert-manager-operator} by adding a `spec.controllerConfig` section in the `CertManager` resource.

.Prerequisites

* You have access to the {product-title} cluster as a user with the `cluster-admin` role.

.Procedure

. Edit the `CertManager` resource by running the following command:
+
[source,terminal]
----
$ oc edit certmanager cluster
----

. Add a `spec.controllerConfig` section with the following override arguments:
+
[source,yaml]
----
apiVersion: operator.openshift.io/v1alpha1
kind: CertManager
metadata:
  name: cluster
  ...
spec:
  ...
  controllerConfig: <1>

>     overrideArgs:

Reviewer: This should be Env instead of Args here.
```suggestion
    overrideEnv:
```

// Module included in the following assemblies:
//
// * security/cert_manager_operator/cert-manager-customizing-api-fields.adoc

:_content-type: PROCEDURE
[id="cert-manager-override-environment-variables_{context}"]

> = Customizing cert-manager Operator API fields by overriding environment variables

Reviewer: Similar here, how about?
`Customizing cert-manager by overriding environment variables from operator API`

You can override the default environment variables for the {cert-manager-operator} by adding a `spec.controllerConfig` section in the `CertManager` resource.

.Prerequisites

* You have access to the {product-title} cluster as a user with the `cluster-admin` role.

.Procedure

. Edit the `CertManager` resource by running the following command:
+
[source,terminal]
----
$ oc edit certmanager cluster
----

. Add a `spec.controllerConfig` section with the following override arguments:
+
[source,yaml]
----
apiVersion: operator.openshift.io/v1alpha1
kind: CertManager
metadata:
  name: cluster
  ...
spec:
  ...
  controllerConfig: <1>
    overrideArgs:
      - 'HTTP_PROXY' <2>
      - 'HTTPS_PROXY' <3>
      - 'NO_PROXY' <4>
----
<1> Add the `spec.controllerConfig` section.
<2>
<3>
<4>

. Save the file to apply the changes.

.Verification

* Verify that the cert-manager pods are running by entering the following command:
+
[source,terminal]
----

> $ oc get pods -n cert-manager

Reviewer: Can we use a yaml output of the pods instead? So users could verify if their newly added args/env have come up in the pod spec.
```sh
$ oc get pods -n cert-manager -o yaml
```
should be good, IMO.

Reviewer: Similar to my the other comment, except for this one you show the environment variables only.
```
$ oc get pods -n cert-manager -o yaml
---
    name: cert-manager-6d48f88495-c88gd
    namespace: cert-manager
---
env:
      - name: POD_NAMESPACE
        valueFrom:
          fieldRef:
            apiVersion: v1
            fieldPath: metadata.namespace
      - name: HTTP_PROXY
---
```

// Module included in the following assemblies:
//
// * security/cert_manager_operator/cert-manager-customizing-api-fields.adoc

:_content-type: PROCEDURE
[id="cert-manager-override-environment-variables_{context}"]
= Customizing cert-manager by overriding environment variables from the cert-manager Operator API

You can override the default environment variables for the {cert-manager-operator} by adding a `spec.controllerConfig` section in the `CertManager` resource.

.Prerequisites

* You have access to the {product-title} cluster as a user with the `cluster-admin` role.

.Procedure

. Edit the `CertManager` resource by running the following command:
+
[source,terminal]
----
$ oc edit certmanager cluster
----

. Add a `spec.controllerConfig` section with the following override arguments:
+
[source,yaml]
----
apiVersion: operator.openshift.io/v1alpha1
kind: CertManager
metadata:
  name: cluster
  ...
spec:
  ...
  controllerConfig: <1>
    overrideEnv:

>       - 'HTTP_PROXY'

Reviewer: I missed it in earlier reviews, the overrideEnv section should be of the format:
```
overrideEnv:
  - name: HTTP_PROXY
     value: http://<PROXY_URL>
  - name: HTTPS_PROXY
     value: https://<PROXY_URL>
   - name: NO_PROXY
     value: <IGNORE_PROXY_DOMAINS>
```
Where <proxy_url> is the url to the proxy server and <ignore_proxy_domains> are comma separated list of domains which proxy should ignore eg. “.example.com,*.local.net”.

xref: https://access.redhat.com/documentation/en-us/openshift_container_platform/3.4/html/installation_and_configuration/install-config-http-proxies

Reviewer 2: Yes, this needs update. https://github.com/openshift/cert-manager-operator/blob/master/deploy/examples/cluster-cert-manager-overrides.yaml shows format example as well

      - 'HTTPS_PROXY'
      - 'NO_PROXY'
----
<1> Add the `spec.controllerConfig` section.

. Save the file to apply the changes.

.Verification

> * Verify that the cert-manager pods are running by entering the following command:

Reviewer 2: Pod will be redeployed after configuring.
```suggestion
* Verify that the cert-manager pods are redeployed by entering the following command:
```

+
[source,terminal]
----

> $ oc get pods -n cert-manager -o yaml
> ----
> +
> .Example output
> [source,yaml]
> ----
> name: cert-manager-6d48f88495-c88gd
> namespace: cert-manager
> ...
> env:
>       - name: POD_NAMESPACE
>         valueFrom:
>           fieldRef:
>             apiVersion: v1
>             fieldPath: metadata.namespace
>       - name: HTTP_PROXY

Reviewer 2: The indentations are not correct. And `POD_NAMESPACE` needs not be documented, because unrelated.
```suggestion
$ oc get pod <redeployed_cert-manager_pod> -n cert-manager -o yaml
----
+
.Example output. The overridden environment variables are added under the `env` field.
[source,yaml]
----
    - name: HTTP_PROXY
      value: http://<PROXY_URL>
    - name: HTTPS_PROXY
      value: https://<PROXY_URL>
    - name: NO_PROXY
      value: <IGNORE_PROXY_DOMAINS>
```

//
// * security/cert_manager_operator/cert-manager-customizing-api-fields.adoc

:_content-type: PROCEDURE
[id="cert-manager-override-environment-variables_{context}"]
= Customizing cert-manager by overriding environment variables from the cert-manager Operator API

You can override the supported environment variables for the {cert-manager-operator} by adding a `spec.controllerConfig` section in the `CertManager` resource.

.Prerequisites

* You have access to the {product-title} cluster as a user with the `cluster-admin` role.
* The {cert-manager-operator} 1.11.1 is installed.

.Procedure

. Edit the `CertManager` resource by running the following command:
+
[source,terminal]
----
$ oc edit certmanager cluster
----

. Add a `spec.controllerConfig` section with the following override arguments:
+
[source,yaml]
----
apiVersion: operator.openshift.io/v1alpha1
kind: CertManager
metadata:
  name: cluster
  ...
spec:
  ...
  controllerConfig: <1>
    overrideEnv:
      - name: HTTP_PROXY
        value: http://<proxy_url> <2>
      - name: HTTPS_PROXY
        value: https://<proxy_url> <2>
      - name: NO_PROXY
        value: <ignore_proxy_domains> <3>
----

> <1> Add the `spec.controllerConfig` section.

Reviewer 3: same comment as in previous module
